src/plugins/data/public/data_sources/datasource_selector/types.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiComboBoxOptionOption, EuiComboBoxSingleSelectionShape } from '@elastic/eui';
+import { GenericDataSource } from '../datasource_services';
+
+export interface DataSourceGroup {
```

Comment:
The naming of these make it sound like their are types that belong to `datasource/types.ts`. We should rename them or move them there.